ui/components/src/EChart.tsx

```diff
@@ -121,62 +102,80 @@ export type OnEventsType<T> = {
 export interface EChartsProps<T> {
   option: EChartsCoreOption;
   sx?: SxProps<Theme>;
-  onChartInitialized?: (instance: ECharts) => void;
   onEvents?: OnEventsType<T>;
   _instance?: React.MutableRefObject<ECharts | undefined>;
-  parentContainer?: HTMLDivElement | null;
+  onChartInitialized?: (instance: ECharts) => void;
 }
 
-export function ECharts<T>({ sx, _instance, onChartInitialized, option, onEvents }: EChartsProps<T>) {
-  const [containerRef, setContainerRef] = useState<HTMLDivElement | null>(null);
-  const [echart, setChart] = useState<ECharts | undefined>(undefined);
-  // Create a chart instance in the container
-  useLayoutEffect(() => {
-    if (containerRef === null) return;
-
-    const chart = init(containerRef);
-    setChart(chart);
-    if (_instance !== undefined) {
-      _instance.current = chart;
+export const EChart = React.memo(function EChart<T>({
+  option,
+  sx,
+  onEvents,
+  _instance,
+  onChartInitialized,
+}: EChartsProps<T>) {
+  const prevOption = useRef<EChartsCoreOption | undefined>();
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const chartElement = useRef<ECharts | null>(null);
+
+  const initChartDom = useCallback(() => {
+    if (containerRef.current !== null && chartElement.current === null) {
+      chartElement.current = init(containerRef.current);
+      onChartInitialized?.(chartElement.current);
+      if (_instance !== undefined) {
+        _instance.current = chartElement.current;
+      }
     }
-    bindEvents(chart, onEvents);
+  }, [_instance, onChartInitialized]);
```

Comment:
Since this function is only used in one place, wonder if it should just be inlined into the spot where it's used?